### DISCUSSION TITLE
Video Public Roadmap & Changelog

### DISCUSSION THREAD

**Discussion by tbarbugli** [GetStream/protocol#127] [Announcements]
## Video Roadmap

We're constantly rearranging this based on customer feedback. Be sure to let us know what's important to you. 

## 1.8 Roadmap - August 2026

* ✅ End-to-end encryption (E2EE) for calls — first on React, mobile SDK support next
* Livestream WebRTC broadcaster mode — optimized go-live from apps at large-audience scale
* Livestream broadcast analytics digest: detailed attendance and publisher/viewer quality reports
* Ringing reliability: server-tracked ring attempts and consistent missed/timeout semantics across platforms
* SIP ingress GA + SIP outgoing calls
* Waiting rooms
* Faster call joins
* ✅ Client-side issues overview in dashboard
* Higher-quality RTMP/SRT ingest transcoding (H.264 High profile, source-true frame rates)
* RTMP with AV1 + fallback codec

## 1.7 Roadmap - March 2026

* ✅ Call attendance stats
* ✅ Realtime stats API for livestreams (fps, bitrate, ...)
* ✅ Raw track recording
* ✅ SIP ingress (beta)
* ✅ Ingress events
* ✅ Simulcast support for WHIP
* ✅ Translated closed-captions (closed beta)

## 1.6 Roadmap - September 2025

* ✅ WHIP
* ✅ Audio hi-fi
* ✅ SRT ingress
* ✅ Python RTC SDK (A/V AI integrations)
* ✅ TURN/STUN on TCP/UDP 443 and static IP ranges

## 1.5 Roadmap - Improved stability, livestreaming, AI Improvements and integrated moderation - June 2025

* ✅ 4k 60fps support for HLS and RTMP
* ✅ Improved client-side telemetry
* ✅ Integrated video moderation 

## 1.4 Roadmap - Livestreaming Improvements - March 2025

Stream scales well to large calls and livestreaming use cases. We can improve which ingress formats we support and make it easier to build large livestreaming experiences:

* ✅ Content: A guide to low latency livestreaming (its easy to cause slowdowns with RTMP, OBS etc. so this needs some additional docs to help our customers)
* ✅ Codec negotiation - dynamically change encoders and decoders during calls
* ✅ Stability improvements, especially in poor network conditions
* ✅ OpenAI realtime integration
* ✅ Java SDK (server-side)
* ✅ Improvements to bandwidth estimation

## 1.3 Roadmap - AI & SIP - Dec 2024

The focus is on improved performance on older devices and poor network conditions. In addition to that we're shipping everything you need to build AI use cases with Stream. (SIP, Python SDK etc.)

* ✅ VP9 as a default over VP8 in more calls
* ✅ Dynascale to improve simulcast/SVC overhead on 1-1 calls 
* ✅ Manual quality selection (currently it's always automatic)
* ✅ Golang SDK
* ✅ Improved Closed Captions

## 1.2 Roadmap - Sept 2024

* ✅ PIP support for React Native
* ✅ PIP support for Flutter
* ✅ Audio only/ Ringing calls document edge cases
* ✅ RTMP out
* ✅ Stability & performance improvements

## 1.1 Roadmap - June 2024

* ✅ Improved support for teams & multi-tenant
* ✅ Improve noise reduction/cancelation
* ✅ Multi language support for transcriptions
* ✅ Complete [moderation endpoints integration](https://getstream.io/video/docs/api/moderation/overview/#banning-users) (blocking users, banning users, ...)
* ✅ Session timers
* ✅ Blur & AI video filter on React Native

## 1.0 Roadmap - April 2024

* ✅ Analytics for calls (this is the most important missing feature during this BETA. When quality drops we need to understand if it's caused by network, CPU, device heat etc.)
* ✅ Blur & AI video filter docs on Android, iOS and React
* ✅ Transcriptions
* ✅ Support external storage for recorded calls (S3, Google Cloud, Azure)
* ✅ Audio & video filter support (supported on native iOS and Android)
* ✅ Screensharing from all mobile SDKs
* ✅ Standardization across SDKs
* ✅ Composite layout for streaming and recording
* ✅ Recording 2.0
* ✅ Unity SDK
* ✅ Livestream, default video player UI for all SDKs
* ✅ Benchmarks for audio rooms and livestreams 
* ✅ Improvements to `starts_at` (allow people to join x minutes earlier)

## After/Later

* Dashboard stats improvements
* PHP SDK
* Expand external storage support to R2, Digital Ocean space and Supabase
* Tap to focus (supported by iOS atm)
* Whiteboards
* Camera controls
* WHEP
* Client-side recording
* Ruby SDK 
* C# SDK (server-side)
* Ingress for SDI, NDI, MTS/ MPEG-2 TS, RIST and Zixi
* Geofencing v2 (User location restrictions, configurable fences, dashboard integration)
* RTSP ingress
* Fixed Streaming Key (RTMP/SRT)

## Video SDK roadmaps

* [React](https://github.com/GetStream/stream-video-js/tree/main/packages/react-sdk#roadmap)
* [iOS](https://github.com/GetStream/stream-video-swift#roadmap)
* [Android](https://github.com/GetStream/stream-video-android/tree/develop#roadmap)
* [Flutter](https://github.com/GetStream/stream-video-flutter/blob/main/development.md)
* [React Native](https://github.com/GetStream/stream-video-js/tree/main/packages/react-native-sdk#Roadmap)

## Changelog

* 17th of August 2026 - Client-side issues overview in dashboard
* 17th of August 2026 - End-to-end encryption (E2EE) for calls, available on React
* 6th of July 2026 - SRT/RTMP ingress events delivered over webhooks
* 4th of June 2026 - Call attendance stats API and realtime livestream stats
* 18th of May 2026 - Automatic audio recovery
* 16th of April 2026 - WHIP OBS simulcast support
* 15th of April 2026 - JoinCall hint to mark a publisher as broadcasting to a large audience, for high-scale livestreams
* 9th of April 2026 - CallKit / Telecom integration for native incoming-call experiences
* 27th of February 2026 - Embeddable pre-built video components for React, and enhanced Picture-in-Picture for iOS
* 20th of January 2026 - Hi-Fi stereo audio output on React Native
* 20th of January 2026 - Individual, Raw and Composite recording support
* 8th of December 2025 - Video Call moderation for React, plus LivestreamLayout improvements (mute, humanized participant names) and audio stats
* 17th of November 2025 - Improved Video Filters and a SpeakerTest component for React
* 10th November 2025 - Support for Hi-Fi and Stereo playback (Android)
* 10th November 2025 - Support for moderation blur and warning events
* 3rd November 2025 - Introduce Telecom support on Android
* 13th October 2025 - Add the possibility to kick a user out of the call
* 30th September 2025 - TURN UDP on well-known IP addr range
* 30th September 2025 - [SRT ingress](https://getstream.io/video/docs/api/streaming/srt/) support
* 30th September 2025 - Audio hi-fi release, support on web and RTMP-ingress, iOS/Android following next
* 22th September 2025 - Released [WHIP ingress](https://getstream.io/video/docs/api/streaming/whip/) 
* 12th of August 2025 - Released new TURN/STUN on TCP/UDP 443 and static IP ranges
* 18th of July 2025 - Inbound Video Pause for React v1.19.0, React Native v1.19.0, and Plain-JS v1.27.0
* 15th of July 2025 - Automatically pause video tracks (congestion control)
* 4th of July 2025 - Agents Python SDK alpha release
* 20th of June 2025 - Improved video telemetry stats is now available on the latest Android SDK 
* 20th of June 2025 - Integrated video moderation is now available out of the box
* 20th of June 2025 - Stereo Audio support for React - v1.18.6, React Native - v1.17.1, Plain-JS - v1.25.0
* 20th of May 2025 - Noise Cancellation support for React Native - v1.14.0
* 13th of May 2025 - Added support for S3 compatible storage (eg. [minIO](https://min.io/))
* 2nd of May 2025 - Improved telemetry and device performance data collection; React v1.15.0, React Native v1.12.0, Plain-JS v1.21.0
* 30th of April 2025 - Added endpoint to [query call participants](https://getstream.io/video/docs/api/calls/#query-call-members)
* 19th of April 2025 - Added support for Riverside RTMP streaming
* 4th of April 2025 - Telemetry data collection; React v1.14.0, React Native v1.11.0, Plain-JS v1.19.0
* 19th of February 2025 - Released official Java SDK
* 19th of February 2025 - HLS end of stream tag support
* 12th of February 2025 - Released a new frame recorder egress that allows customers to receive keyframe image over webhooks for live calls
* 27th of January 2025 - Major HLS latency improvements, average delay is reduced from 20 seconds down to 10 seconds
* 15th of January 2025 - Codec Negotiation support in React v1.10.0, React Native v1.8.0, and Plain-JS v1.15.0
* 2nd of January 2025 - Closed Captions support in React v1.9.0, React Native v1.6.0, and Plain-JS v1.14.0
* 24th December 2024 - Closed captions support
* 24th December 2024 - Improve speech-to-text accuracy and reduce hallucinations in transcriptions
* 16th of October 2024 - VP9 and AV1 support is available with React (1.7.0), React Native (1.2.0), and Plain-JS (1.9.0)
* 16th of October 2024 - Dynascale to remove simulcast/SVC on 1-1 calls is available with React (1.7.0), React Native (1.2.0), and Plain-JS (1.9.0)
* 4th of October 2024 - Golang SDK is publicly available 
* 2nd of October 2024 - Manual Quality Selection is available with React (1.5.0), React Native (1.1.0), and Plain-JS (1.8.0)
* 3rd of September 2024 - React Native v1.0.0 is now released
* 3rd of September 2024 - Reconnects v2 for React (v1.3.0+), React Native (v1.0.0) and Plain-JS (v1.6.0)
* 28th of August 2024 - Picture in Picture (PiP) support for React Native iOS (v0.10.5)
* 16th of August 2024 - RTMP out is now supported
* 19th of July 2024 - Picture in Picture (PiP) support on Flutter SDK
* 12th of June 2024 - Blur and AI Video filters for React Native (Android, iOS)
* 10th June 2024 - Moderation endpoints (blocking users, banning, muting, ...)
* 31st May 2024 - [Session timers API](https://getstream.io/video/docs/api/calls/#session-timers) are available
* 14th of May 2024 - [User bans](https://getstream.io/video/docs/api/moderation/overview/#banning-users) work across video and chat products
* 10th of May 2024 - Released Python SDK v1.0.0
* 7th of May 2024 - [React 1.0.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-1.0.0) released, [Plain-JS 1.0.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-client-1.0.0) released
* 1st of May 2024 - iOS 1.0 release, https://github.com/GetStream/stream-video-swift/releases/tag/1.0.0
* 26th of April 2024 - [Improved support for teams & multi-tenant](https://getstream.io/video/docs/api/multi-tenant/) and Analytics for calls is GA on Stream Dashboard
* 25th of April 2024 [React Video SDK 0.6.12](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-0.6.12), [Plain-JS SDK 0.7.8](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-client-0.7.8) - Noise Cancellation
* 9th of April 2024: [React Video SDK 0.6.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-0.6.0), [React Native Video SDK 0.6.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-native-sdk-0.6.0), [Plain-JS Video SDK 0.7.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-client-0.7.0) - Removes the already deprecated server-side support in favor of our [NodeJS Server-Side SDK](https://github.com/GetStream/stream-node)
* 5th of April 2024: [React Video SDK 0.5.10](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-0.5.10), [Plain-JS Video SDK 0.6.8](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-client-0.6.8) - Background blurring and background filters
* 2nd of April 2024: [Flutter Video SDK v0.3.5](https://github.com/GetStream/stream-video-flutter/releases/tag/v0.3.5): Picture in Picture for Android
* 25th of March 2024: [React Video SDK 0.5.5](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-0.5.5), [React Native Video SDK 0.5.10](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-native-sdk-0.5.10), [Plain-JS Video SDK 0.6.3](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-client-0.6.3) - Call statistics reporting
* 26th of February 2024: [React Video SDK 0.5.0 release](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-0.5.0)
* 26th of February 2024: [React Native Video SDK 0.5.0 release](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-native-sdk-0.5.0)
* 22nd of February 2024: Transcriptions went live
* 16th of February 2024: iOS 0.5 release, https://github.com/GetStream/stream-video-swift/releases/tag/0.5.0
* 9th of February 2024: Support external storage for recorded calls (S3, Google Cloud, Azure) is available
* 9th of February 2024: [Flutter Video SDK v0.3.2](https://github.com/GetStream/stream-video-flutter/releases/tag/v0.3.2): fast reconnect, background voip callback for iOS terminated state, push notifications enhancements
* 3rd of January 2024: Audio & video filter support on iOS and Android, Screensharing is supported for all mobile SDKs
* 12th of December 2023: [Flutter Video SDK v0.3.0](https://github.com/GetStream/stream-video-flutter/releases/tag/v0.3.0)
* 13th of November 2023: Python and Node.JS server-side SDK are [live and documented](https://getstream.io/video/docs/api/)
* 10th of November 2023: [Unity SDK is GA](https://getstream.io/video/docs/unity/), recording 2.0 is live, support for composite layout on streaming and recording is also live
* 27th of October 2023: [React Video SDK v0.4.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-react-sdk-0.4.0) and [Plain-JS SDK v0.4.0](https://github.com/GetStream/stream-video-js/releases/tag/%40stream-io%2Fvideo-client-0.4.0) released
* 24th of October 2023, Android SDK 0.4.0 released, https://github.com/GetStream/stream-video-android/releases/tag/0.4.0
* 11th of October 2023: iOS 0.4 release, https://github.com/GetStream/stream-video-swift/releases/tag/0.4.0
* 06th of October 2023: React SDK v0.3.38: Added support for Screen Share Audio
* 06th of October 2023: Plain-JS SDK v0.3.30: Added support for Screen Share Audio
* 28th of September 2023: Flutter SDK v0.1.0: Improved VoIP and CallKit support, SDK alignment, Docs: https://github.com/GetStream/stream-video-flutter/releases/tag/v0.1.0
* 27th of September 2023: React SDK v0.3.31: LivestreamLayout and support for thumbnails 
* 22nd of September 2023: Thumbnails for livestreaming previews
* 21nd of September 2023: Android 0.3.4, fast reconnect: https://github.com/GetStream/stream-video-android/releases/tag/0.3.4
* 15th of September 2023: Video call benchmark up to 10k participants (goal is 100k by end of Q4)
* 8th of September 2023: Expo support
* 3rd of September 2023: React native release
* 25th of August 2023: iOS 0.3 release, https://github.com/GetStream/stream-video-swift/releases/tag/0.3.0
* 16th of August 2023: React v0.3, JS-SDK 0.3: Improved Call State handling and new Device Management API
* 7th of August 2023: RTMP supports multiple resolutions and audioLevels
* 7th of August 2023: Android 0.3 release (ringing, livestreaming and chat improvements): https://github.com/GetStream/stream-video-android/releases/tag/0.3.0
* 4th of August 2023: Improvements to RTMP support, audio levels, simulcast and others
* 4th of August 2023: Benchmark up to 400 video call participants in a single call. Blogpost coming later
* 4th of August 2023: Livestream tutorials
* 20th of July 2023: Android 0.2 release, https://github.com/GetStream/stream-video-android/releases
* 18th of July 2023: iOS 0.2 release, https://github.com/GetStream/stream-video-swift/releases/tag/0.2.0
* 10th of July 2023: Kotlin, Swift, React and Flutter 0.1 release

**Comment by caaazinha** [comment]
hi! our company is using GetStream chat and video solution but we do have an use case for video where the "Waiting rooms" feature would be extremely important and totally nail... any chance of being released in this years' roadmap?

**Reply by tbarbugli** [reply]
> hi! our company is using GetStream chat and video solution but we do have an use case for video where the "Waiting rooms" feature would be extremely important and totally nail... any chance of being released in this years' roadmap?

Right now we only support waiting room partially with the backstage functionality. We are planning to fully support this functionality later in 2025
